### PR TITLE
Add search index method functionality

### PR DESCRIPTION
**Pull request by ComradeTA**

### CODE REVIEW

File: src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java [hunk 1]
```diff
@@ -12,7 +16,34 @@
...
+                finalQuery += "AND token = ?";
+            }
+
+            PreparedStatement queryStatement =DBConn.prepareStatement(finalQuery);
+
+            for (int i = 0; i < tempList.size(); i++) {
+                queryStatement.setString(i+1, String.valueOf(tempList.get(i)));
+            }
+
+            ResultSet
```
Comment (OthelloEngineer) [inline src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java]: ResultSet???

**Review by OthelloEngineer — CHANGES_REQUESTED** [review]
You're making a format for a query,
trying an SQLquery,
preparing a statement,
preparing another nested statement, 
executing the statement, and
formatting your ResultSet.

Use method extraction and consider calling the same extracted method when preparing your second statement. 

“In order to make sure our functions are doing 'one thing', we need to make sure that the statements within the function are all at the same level of abstraction.” - Uncle Bob (Clean code)

File: src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java [hunk 2]
```diff
@@ -12,7 +18,43 @@
...
     @Override
     public List<Brand> searchBrandIndex(List<String> tokens) {
-        return null;
+        List<Brand> brandList = new ArrayList<>();
+        List<Integer> tempList = new ArrayList<>();
+        String finalQuery = "SELECT * FROM TokenBrandMap LEFT JOIN Brand ON Brand.id = brandId WHERE tokenId = ?";
+
+        try {
+            for (int i = 0; i < tokens.size(); i++) {
+                PreparedStatement queryStatement = DBConn.prepareStatement("SELECT id FROM Tokens WHERE token = ?");
```
Comment (mathiasgredal) [inline src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java:27]: Prepared statement is never closed

File: src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java [hunk 3]
```diff
@@ -12,7 +18,43 @@
...
+                queryStatement.setString(1, tokens.get(i));
+                ResultSet queryResultSet = queryStatement.executeQuery();
```
Comment (mathiasgredal) [inline src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java:29]: Resultset is never closed

File: src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java [hunk 4]
```diff
@@ -12,7 +18,43 @@
...
+                tempList.add(Integer.valueOf(queryResultSet.getString("id")));
+            }
+
+            for (int i = 0; i < tempList.size(); i++) {
+                finalQuery += " AND token = ? ";
+            }
+
+            PreparedStatement queryStatement = DBConn.prepareStatement(finalQuery);
```
Comment (mathiasgredal) [inline src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java:37]: This prepared statement is not closed either

File: src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java [hunk 5]
```diff
@@ -12,7 +18,43 @@
...
+
+            for (int i = 0; i < tempList.size(); i++) {
+                queryStatement.setString(i + 1, String.valueOf(tempList.get(i)));
+            }
+
+            ResultSet queryResultSet = queryStatement.executeQuery();
```
Comment (mathiasgredal) [inline src/main/java/dk/sdu/se_f22/brandmodule/index/BrandIndex.java:43]: This resultset is not closed

**Review by mathiasgredal — CHANGES_REQUESTED** [review]
You have to close prepared statements and result sets. This is preferably done using try-with-resources.

**Pull request merged**